libraries/adaptive-expressions/src/expressionFunctions.ts

```diff
@@ -1793,6 +1793,25 @@ export class ExpressionFunctions {
         return count;
     }
 
+    private static flatten(arr: any[], dept: number): any[]{
+        dept = typeof dept === 'undefined' ? 1 : dept;
```

Comment:
> ? 1  [](start = 42, length = 5)

This changes the meaning.  No depth should flatten all arrays, otherwise to depth.  This is different than the flat function, but is a better definition.